draft-thomson-httpbis-h2-stream-limits.md

```diff
@@ -53,18 +57,23 @@ of abuse.
 
 # Introduction
 
-HTTP/2 {{RFC9113}} allows endpoints to describe a concurrency limit for streams
-(`SETTINGS_MAX_CONCURRENT_STREAMS`).  Initially, the stream concurrency limit is
-only bounded by the maximum number of streams that can be created, which is
-2<sup>30</sup> for each endpoint, but the value can be changed at any time.
-Most endpoints set a smaller value in an attempt to protect the resources that
-are committed when processing a stream.
-
-This limit is not effective in the case that streams are quickly cancelled.
-Stream cancellation with the `RST_STREAM` frame has immediate effect, which
-means that the stream no longer counts against the concurrent stream limit.
-This means that a malicious endpoint can create and cancel an unbounded number
-of streams as long as its peer does not set a limit of zero.
+HTTP/2 allows endpoints to declare the concurrency limit for streams created by
+their peer (`SETTINGS_MAX_CONCURRENT_STREAMS`; see {{Section 5.1.2 of
+RFC9113}}). Initially, the stream concurrency limit is only bounded by the
+maximum number of streams that can be created, which is 2<sup>30</sup> for each
+endpoint, but the value can be changed at any time. Most endpoints set a smaller
+value in an attempt to protect the resources that are committed when processing
+a stream.
+
+This limit is not effective in the case that streams are quickly cancelled or
+closed. The creator of a stream can cancel it using the `RST_STREAM` frame.  The
+creator of a stream can also purposefully send frames that violate HTTP/2 rules,
+forcing a compliant receiver to close the stream. Either of these methods has
+deterministic and immediate effect, which means that the stream no longer counts
+against the concurrent stream limit.  This means that a malicious endpoint can
+create or cancel an unbounded number of streams as long as its peer does not set
+a limit of zero.  For clarity, we'll refer to either of these methods as
+"cancelling".
```

Comment:
Didyou intend to roll up the other PR here?  We can do PRs in parallel.

Reply:
Sorry no, I meant to split them but completely bollocksed it up. Force pushed to drop the unrelated commit.